src/main/java/org/overture/ego/security/JWTAccessToken.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2017. The Ontario Institute for Cancer Research. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.overture.ego.security;
+
+import io.jsonwebtoken.Claims;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.overture.ego.token.TokenService;
+import org.springframework.security.oauth2.common.OAuth2AccessToken;
+import org.springframework.security.oauth2.common.OAuth2RefreshToken;
+
+import java.util.*;
+
+@Slf4j
+@Data
+public class JWTAccessToken implements OAuth2AccessToken {
+
+  Claims tokenClaims = null;
+  String token= null;
+  public JWTAccessToken(String token, TokenService tokenService){
+    this.token = token;
+    this.tokenClaims = tokenService.getTokenClaims(token);
+  }
+
+  @Override
+  public Map<String, Object> getAdditionalInformation() {
+    val output = new HashMap<String, Object>();
+    output.put("groups",getUser().get("groups"));
+    return output;
+  }
+
+  @Override
+  public Set<String> getScope() {
```

Comment:
```java
import static com.google.common.collect.Sets.newHashSet;



  @Override
  public Set<String> getScope() {
    return newHashSet((ArrayList<String>)getUser().get("roles"));
  }
```

Though to be honest, I'm seeing a lot of `Map<String, Object>` in the code. Does it make sense to create `Value` classes to handle this so we are least get type safety? 

A new developer coming in seeing `Map<String,Object>` being thrown around in the code won't know what to expect of use unless they drop down into a debugger.